### ISSUE TITLE
Panic while marshaling protobuf messages 

### ISSUE THREAD

**Issue by creachadair** [tendermint/tendermint#7767]
Context: After the merge from ABCI++ in 39ffa80ae77a927f5236a4bb0380c5ac5254f9bb, end-to-end tests are failing in CI. For example: https://github.com/tendermint/tendermint/runs/5047090890

Example log trace:

```
validator01    | panic: runtime error: invalid memory address or nil pointer dereference
validator01    | [signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x7b26cc]
validator01    | 
validator01    | goroutine 108 [running]:
validator01    | github.com/tendermint/tendermint/proto/tendermint/types.(*Vote).MarshalToSizedBuffer(0x0, {0xc001cb4002, 0x2190, 0x2198})
validator01    | 	github.com/tendermint/tendermint/proto/tendermint/types/types.pb.go:1557 +0x2c
validator01    | github.com/tendermint/tendermint/abci/types.(*RequestPrepareProposal).MarshalToSizedBuffer(0xc001ca6600, {0xc001cb4002, 0x2190, 0x2198})
validator01    | 	github.com/tendermint/tendermint/abci/types/types.pb.go:5183 +0x30d
validator01    | github.com/tendermint/tendermint/abci/types.(*Request_PrepareProposal).MarshalToSizedBuffer(0x3, {0xc001cb4002, 0x2190, 0xa2beae})
validator01    | 	github.com/tendermint/tendermint/abci/types/types.pb.go:4549 +0x3c
validator01    | github.com/tendermint/tendermint/abci/types.(*Request_PrepareProposal).MarshalTo(0xa2beae, {0xc001cb4002, 0xc000013dd0, 0x2198})
validator01    | 	github.com/tendermint/tendermint/abci/types/types.pb.go:4542 +0x47
validator01    | github.com/tendermint/tendermint/abci/types.(*Request).MarshalToSizedBuffer(0xc001ca0550, {0xc001cb4002, 0x2190, 0x2198})
validator01    | 	github.com/tendermint/tendermint/abci/types/types.pb.go:4238 +0x99
validator01    | github.com/tendermint/tendermint/abci/types.(*Request).MarshalTo(0x11df1e0, {0xc001cb4002, 0xc001ca0550, 0x2198})
validator01    | 	github.com/tendermint/tendermint/abci/types/types.pb.go:4226 +0x47
validator01    | github.com/tendermint/tendermint/internal/libs/protoio.(*varintWriter).WriteMsg(0xc000013ea8, {0x14c7370, 0xc001ca0550})
validator01    | 	github.com/tendermint/tendermint/internal/libs/protoio/writer.go:63 +0x27e
validator01    | github.com/tendermint/tendermint/abci/types.WriteMessage({0x14c7370, 0xc001ca0550}, {0x14b57a0, 0xc000080c00})
validator01    | 	github.com/tendermint/tendermint/abci/types/messages.go:17 +0xb8
validator01    | github.com/tendermint/tendermint/abci/client.(*socketClient).sendRequestsRoutine(0xc00017ad10, {0x14d8c10, 0xc0002b4080}, {0x7f91ccad9fd8, 0xc0002607f8})
validator01    | 	github.com/tendermint/tendermint/abci/client/socket_client.go:141 +0x32d
validator01    | created by github.com/tendermint/tendermint/abci/client.(*socketClient).OnStart
validator01    | 	github.com/tendermint/tendermint/abci/client/socket_client.go:89 +0x[297](https://github.com/tendermint/tendermint/runs/5055048664?check_suite_focus=true#step:6:297)
```

**Comment by creachadair** [comment]
Thanks to @sergio-mena for adding context here—it looks like maybe we will need to do some additional work to update the protobuf generation.

**Comment by sergio-mena** [comment]
Addressed via #7821
